libpod/runtime_ctr.go

```diff
@@ -194,6 +224,14 @@ func (r *Runtime) removeContainer(ctx context.Context, c *Container, force bool)
 		// Lock the pod while we're removing container
 		pod.lock.Lock()
 		defer pod.lock.Unlock()
+		if err := pod.updatePod(); err != nil {
+			return err
+		}
+
+		infraID := pod.state.InfraContainerID
+		if c.ID() == infraID {
+			return errors.Errorf("an infra container cannot be removed without removing pod %s", pod.ID())
```

Comment:
Suggest rewording to "container %s is the infra container of pod %s and cannot be removed without removing the pod"